File: media-service/src/main/java/com/mordent/ua/mediaservice/model/data/SongMetadata.java
package com.mordent.ua.mediaservice.model.data;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Map;

@Document
public record SongMetadata(
    @Id Long id,
    Map<String, String> metadata
) {
}

File: metadata-service/src/main/kotlin/com/mordent/ua/metadataservice/model/data/SongMetadata.kt
package com.mordent.ua.metadataservice.model.data

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document
class SongMetadata(
    @Id
    val id: Long,
    val metadata: Map<String, String>
) {
}